hack/dev/Dockerfile.okteto

```diff
@@ -0,0 +1,176 @@
+FROM golang:1.17 as builder
```

Comment:
I think we should make part of these a permanent image in a shared registry somewhere.  I don't think it would need to hold this back, but more of a thought on next steps.  I'm experimenting on some like that now, mixed with some other optimizations.